Pau/ci trivy tmp2

pau-hedgehog opened the pull request.

pau-hedgehog commented:
Sarif consolidation matches Security tab:
<img width="675" height="1130" alt="image" src="https://github.com/user-attachments/assets/fbc0e00f-bfd1-43f9-89df-6ae99bfdbc39" />

<img width="1746" height="1132" alt="image" src="https://github.com/user-attachments/assets/c8fe2ccd-90d1-4335-906d-4861635d5df6" />

pau-hedgehog commented:
In a fresh PR the scan results match the Security tab counts (check `Total vulnerability instances`):
<img width="656" height="1060" alt="image" src="https://github.com/user-attachments/assets/4cc92611-39df-4ce0-a7b9-41ae280cc24a" />
VS
<img width="2040" height="1108" alt="image" src="https://github.com/user-attachments/assets/5953fe40-de12-42ab-808f-628e51c126a1" />

If you run a scan for all VM types:
<img width="2275" height="676" alt="image" src="https://github.com/user-attachments/assets/51c19c84-6110-442d-a957-8ab3aef6a775" />

The Security tab gets updated properly:
<img width="651" height="1231" alt="image" src="https://github.com/user-attachments/assets/2ca64880-a8fe-4f4d-992b-eeeb15fa921e" />
VS
<img width="2027" height="1090" alt="image" src="https://github.com/user-attachments/assets/7c27e8f2-4f55-4cad-a2f7-9b767824df70" />

pau-hedgehog commented:
If you run a scan with a narrower scope (default), the findings are updated properly:
<img width="2027" height="1090" alt="image" src="https://github.com/user-attachments/assets/d3b21079-f288-4483-9a99-9ed5dfec04de" />

edipascale commented on hack/vlab-trivy-runner.sh:
```diff
@@ -697,155 +509,106 @@ else
+                echo "Assigning ${#images_for_switch[@]} images to $switch: ${images_for_switch[*]}"
+
+                (
+                    echo "Starting load-balanced scan on $switch..."
```
@pau-hedgehog I think this is the only potentially relevant comment from Copilot. I don't think we should be running multiple scans concurrently, but if I understand the workflow yaml we currently allow this for different branches, and we could run into this issue, is this correct?

pau-hedgehog commented:
> [@pau-hedgehog](https://github.com/pau-hedgehog) I think this is the only potentially relevant comment from Copilot. I don't think we should be running multiple scans concurrently, but if I understand the workflow yaml we currently allow this for different branches, and we could run into this issue, is this correct?

I saw this comment and I realized it was overcomplicated. Also it had a bug which didn't show all the vulns of the SONiC VM (a ton). Fixed it and I've kept it as option because it's half the (scan/process) time but the default scan mode is single-switch now.

I've fixed the local mode of running this also, which could enable use to run as part of the VLAB run and save us the precious `vlab up` time.
